### PR TITLE
Allow custom docs for Weight enums and Weights fields

### PR DESCRIPTION
**Pull request by NicolasHug**
This PR makes it possible to document a weight enum, or a single `Weights` class.

rendered docs: https://output.circle-artifacts.com/output/job/cc788413-d4e6-4e1b-be75-453a8b5fb736/artifacts/0/docs/models/generated/torchvision.models.resnet50.html#torchvision.models.ResNet50_Weights

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -335,10 +342,14 @@ def inject_weight_metadata(app, what, name, obj, options, lines):
             metrics = meta.pop("metrics", {})
             meta_with_metrics = dict(meta, **metrics)
 
+            meta_with_metrics.pop("categories", None)  # We don't want to document these, they can be too long
```
Comment (NicolasHug) [inline docs/source/conf.py:345]: This is mostly unrelated, I can revert.

Comment (datumbox) [reply docs/source/conf.py:345]: It's cool with me.

File: torchvision/models/resnet.py [hunk 2]
```diff
@@ -344,6 +344,7 @@ class ResNet34_Weights(WeightsEnum):
 
 
 class ResNet50_Weights(WeightsEnum):
+    """<Some doc specific to ResNet50_Weights>"""
```
Comment (datumbox) [inline torchvision/models/resnet.py:347]:
```suggestion
    """Below you can find the available pre-trained weights for ResNet50:"""
```

File: torchvision/models/resnet.py [hunk 3]
```diff
@@ -355,6 +356,10 @@ class ResNet50_Weights(WeightsEnum):
                 "acc@1": 76.130,
                 "acc@5": 92.862,
             },
+            # Putting this here but the _docs key could also just be a field of the Weights dataclass
+            "_docs": """
+            <Some doc specific to ResNet50_Weights.IMAGENET1K_V2>
```
Comment (datumbox) [inline torchvision/models/resnet.py:361]:
```suggestion
            Basic training recipe for ResNet50:
```

File: torchvision/models/resnet.py [hunk 4]
```diff
@@ -368,6 +373,9 @@ class ResNet50_Weights(WeightsEnum):
                 "acc@1": 80.858,
                 "acc@5": 95.434,
             },
+            "_docs": """
+            <Some doc specific to ResNet50_Weights.IMAGENET1K_V2>
```
Comment (datumbox) [inline torchvision/models/resnet.py:377]:
```suggestion
            Improved weights using TorchVision's `new recipe <https://pytorch.org/blog/how-to-train-state-of-the-art-models-using-torchvision-latest-primitives/>`_:
```

**Review by datumbox — APPROVED** [review]
Looks great to me. I've proposed some "better" dummy text so that we can merge the PR instead of leaving it as a POC. Can come back later and put more meaningful text.

**Comment by datumbox** [comment]
The schema test is [failing](https://app.circleci.com/pipelines/github/pytorch/vision/17482/workflows/4d0b442e-46c4-4bc4-acf6-ac42ba873d6f/jobs/1415410) and it's related. We just need to add `_docs` on the [permitted](https://github.com/pytorch/vision/blob/main/test/test_extended_models.py#L83) fields.

**Pull request merged**